lstchain/reco/dl0_to_dl1.py

```diff
@@ -312,33 +312,38 @@ def r0_to_dl1(input_filename = get_dataset_path('gamma_test_large.simtel.gz'),
                     dl1_container.gps_time = event.trig.gps_time.value
 
                     if not is_simu:
-                        # GPS + WRS + UCTS is now working in its nominal configuration. 
+                        # GPS + WRS + UCTS is now working in its nominal configuration.
                         # These TS are stored into ucts_time container.
-                        # TS can be alternatively calculated from the TIB/Dragon 
-                        # modules counters + NTP time corresponding to the start 
-                        # of the run (just for cross-checks). For the time being,
-                        # the three TS will be stored in the DL1 files.
-                        # This will be deprecated and modified back to uniquely use
-                        # gps_time whenever the GPS + WRS + UCTS info reliably and stably
-                        # arrives at the EvB side.
+                        # TS can be alternatively calculated from the TIB and Dragon module
+                        # counters by using the first UCTS TS as the reference point.
+                        # For the time being, the three TS are stored in the DL1 files
+                        # for checking purposes.
 
                         # gps_time = event.r0.tel[telescope_id].trigger_time
 
-                        ucts_time = event.lst.tel[telescope_id].evt.ucts_timestamp * 1e-9 # nsecs
+                        ucts_time = event.lst.tel[telescope_id].evt.ucts_timestamp * 1e-9  # nsecs
 
-                        # Get counters from the central Dragon module
-                        module_id = 82
+                        module_id = 82  # Get counters from the central Dragon module
 
-                        dragon_time = (
-                                event.lst.tel[telescope_id].svc.date +
+                        if i == 0:
```

Comment:
Hi, at the moment this will be looking into the first event of the file you are providing, that will only be the first event of the run in case you are processing the first subrun.